### ISSUE TITLE
[QUESTION] Blade accessing Entity protected property

### ISSUE THREAD

**Issue by TerraTech** [laravel-doctrine/orm#337]
### Package version, Laravel version
Laravel: 5.6
LaravelDoctrine: 1.4.7

Tonight was my first time using laraveldoctrine and the initial (newbie) learning curve was a bit steep just in getting the Auth functionality to work correctly.  e.g. getting hit with the "Class users does not exist" problem.  NOTE: I had used "php artisan make:auth" to set it up.

After a bit of determination I ended up getting it to work correctly once I understood more about what it was looking for and adding what was missing.

FWIW, 'name', 'email', 'created_at', 'updated_at' should be in the 'LaravelDoctrine\ORM\Auth\Authenticable' trait or at least a note in the docs about it for first timers.  Setting up 'Auth' is probably one of the first steps that many will choose to do when learning Laravel and LaravelDoctrine.

My question is in regards to how the Blade system retrieves properties from the entity class.  In particular:
app.stub ==> app.blade.php
```php
{{ Auth::user()->name }}
```
I had that property set as:
```php
/** @ORM\Column(type="string") */
protected $name;
```
Unfortunately, that won't work because Blade is accessing it directly, therefore it was wanting it set to '_public_'.  I make it a general rule to never have '_public_' properties in a Class, so I did some tinkering and switched it back to '_protected_', added a getter:
```php
public function getName(): string {return $this->name;}
```
and changed the Blade accessor to use the entity's new getter:
```php
{{ Auth::user()->getName() }}
```
Which worked as desired.

Unfortunately, that was only one class property, and I feel like I am running headfirst into a huge minefield as I develop and add more entities.  It seems that using 'php artisan make;*' is going to use boilerplate stub templates and lob more hand grenades my way pushing me to surrender and use 'public' properties in the entities which I will vehemently resist to the point of walking away from it or just using its baked in Eloquent.  I really like the separation that Doctrine ORM provides and really want to find a way to cleanly solve this.

So, before I plan on what my next steps are, I feel like I am overlooking something - somewhere that is to bridge this divide.  Is there a configuration setting or secret sauce somewhere that will allow Blade to access an entity's _protected_ property via a getter/setter?

Thank you for your time reading through this and whatever assistance you can provide.

**Comment by jcaillot** [comment]
required accessors (getter and setter) can be generated using doctrine:generate:entities.

**Comment by TerraTech** [comment]
I've figured it out and many things have become much clearer now after climbing the learning curve a bit more.  Overall I decided to just let Auth ride with Eloquent since I'm not extending or customizing the login handling.  If I do have to extend it in the future I know what to do now.

@jcaillot that wouldn't really work because it fails when it hits the `password_resets` table since it doesn't have a primary key.
